src/rdfTypes/Variable.cpp

```diff
@@ -61,19 +63,51 @@ Variable Variable::getWordScoreVariable(std::string_view word,
 
 // _____________________________________________________________________________
 Variable Variable::getMatchingWordVariable(std::string_view term) const {
-  return Variable{
-      absl::StrCat(MATCHINGWORD_VARIABLE_PREFIX, name().substr(1), "_", term)};
+  auto s = absl::StrCat(MATCHINGWORD_VARIABLE_PREFIX, name().substr(1), "_");
+  appendEscapedWord(term, s);
+  return Variable{std::move(s)};
 }
 
+namespace {
+// Returns true for a  subset of characters that are valid in variable names.
+// This roughly corresponds to `PN_CHARS_BASE` from the SPARQL 1.1 grammar with
+// the characters 0-9 also being allowed. Note that this deliberately does not
+// contain the (valid) character `_`, as we use that character for escaping and
+// it thus has to be escaped itself.
+constexpr bool codePointSuitableForVariableName(UChar32 cp) {
+  constexpr auto validRanges = []() {
+    using A = std::array<UChar32, 2>;
+    return std::array{
+        A{'A', 'Z'},       A{'a', 'z'},       A{'0', '9'},
+        A{0x00C0, 0x00D6}, A{0x00D8, 0x00F6}, A{0x00F8, 0x02FF},
+        A{0x0370, 0x037D}, A{0x037F, 0x1FFF}, A{0x200C, 0x200D},
+        A{0x2070, 0x218F}, A{0x2C00, 0x2FEF}, A{0x3001, 0xD7FF},
```

Comment:
I have refactored this for better coverage
(the nested `||` want to cover all branches...).